ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -160,29 +187,92 @@ internal class CustomerCenterViewModelImpl(
         }
     }
 
-    private suspend fun loadPurchaseInformation(): PurchaseInformation? {
+    private suspend fun loadPurchaseInformation(
+        dateFormatter: DateFormatter,
+        locale: Locale,
+    ): PurchaseInformation? {
         val customerInfo = purchases.awaitCustomerInfo(fetchPolicy = CacheFetchPolicy.FETCH_CURRENT)
 
-        // Customer Center WIP: update when we have subscription information in CustomerInfo
-        val activeEntitlement = customerInfo.entitlements.active.values.firstOrNull()
-        if (activeEntitlement != null) {
-            val product = purchases.awaitGetProduct(activeEntitlement.productIdentifier).first()
-            val locale = getDefaultLocales().first()
-            val purchaseInformation = PurchaseInformation(
-                title = product.description,
-                durationTitle = product.period?.localizedPeriod(locale) ?: "",
-                price = product.price.formatted,
-                expirationDateString = activeEntitlement.expirationDate.toString(),
-                willRenew = activeEntitlement.willRenew,
-                active = activeEntitlement.isActive,
-                productId = activeEntitlement.productIdentifier,
-            )
-            return purchaseInformation
+        val hasActiveProducts = customerInfo.activeSubscriptions.isNotEmpty() ||
+            customerInfo.nonSubscriptionTransactions.isNotEmpty()
+
+        if (hasActiveProducts) {
+            val activeTransactionDetails = findActiveTransaction(customerInfo)
+
+            if (activeTransactionDetails != null) {
+                val entitlement = customerInfo.entitlements.all.values
+                    .firstOrNull { it.productIdentifier == activeTransactionDetails.productIdentifier }
+
+                return createPurchaseInformation(activeTransactionDetails, entitlement, dateFormatter, locale)
+            } else {
+                Logger.w("Could not find subscription information")
+            }
         }
 
         return null
     }
 
+    private fun findActiveTransaction(customerInfo: CustomerInfo): TransactionDetails? {
```

Comment:
So for now we will only display info of one transaction right? Not all of them if there are multiple?

Reply:
yes, just one, like in iOS. In iOS we've added a link to see the rest of your purchases though, and that will come to Android as well